src/main/java/com/puppycrawl/tools/checkstyle/checks/design/FinalClassCheck.java

```diff
@@ -168,6 +182,17 @@ public void visitToken(DetailAST ast) {
                 }
                 break;
 
+            case TokenTypes.LITERAL_NEW:
+                if (ast.getFirstChild() != null) {
+                    for (ClassDesc classDesc : classes) {
+                        if (ast.getLastChild().getType() == TokenTypes.OBJBLOCK
```

Comment:
`ast` check is in `for` loop, but it's value will not change between the loops.
This should be added into `if (ast.getFirstChild() != null) {` check instead. It only needs to be checked once.

Reply:
Sry my bad, done.